src/vim-loader/remoteValue.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Returns a value from cache or queue up existing request or start a new requests
+ */
+export class RemoteValue<T> {
+  label: string
+  private _value: T
+  private _getter: () => Promise<T>
+  private _request: Promise<T>
+
+  constructor (getter: () => Promise<T>, label?: string) {
+    this._getter = getter
+    this.label = label
+  }
+
+  get (): Promise<T> {
+    if (this._value) {
+      // console.log(this.label + ' returning cached value ')
+      return new Promise((resolve, reject) => resolve(this._value))
```

Comment:
Code golf 🏌️‍♀️: we can use `Promise.resolve(this._value)` as shorthand here.

Reply:
Oh cool, I wondered how to do that.